# 🪵🍁🍂🍄🍗🦃🏈 Big Thanksgiving Dinner Analysis 🏈🦃🍗🍄🍂🍁🪵
**Name:** Austin Norman       <br>
**Class:** Section 1

In [1]:
# Libraries
import numpy as np
import pandas as pd

In [2]:
# Synthetic Thanksgiving Data
np.random.seed()    # set seed to 2310
n = 50    # set n to 50

# creating the variables first (Age, Dinner, Dessert, Duration)
Age = (np.random.normal(37, 18, n).round(0).clip(5, 90))
Num_Dinner_Servings = np.round(4 - 0.0009 * (Age - 40)**2 + np.random.normal(0, 0.3, n)).clip(1, 5)
Num_Dessert_Servings = np.round(np.clip(2.5 - (Age / 50) + np.random.normal(0, 0.5, n), 0, 3))
Dinner_Duration_Hours =  np.round(1.5 + (Age / 40) + np.random.normal(0, 0.5, n), 1).clip(1, 5)

# creating boolean for whether or not they watched tv (older and younger people more likely to watch)
prob_parade = np.clip(    0.3 + 0.4 * np.exp(-((Age - 15)/20)**2) + 0.4 * np.exp(-((Age - 70)/20)**2),0, 1)
Watched_Parade = np.random.binomial(1, prob_parade)

# creating the data frame using the variables we just made
Thanksgiving = pd.DataFrame({
    'Ages' : Age,
    'Dinner_Servings' : Num_Dinner_Servings,
    'Dessert_Servings' : Num_Dessert_Servings,
    'Duration' : Dinner_Duration_Hours,
    'Macy_Parade' : Watched_Parade
})

Thanksgiving.head()


Ages  Dinner_Servings  Dessert_Servings  Duration  Macy_Parade
0  33.0              4.0               2.0       2.6            0
1  47.0              4.0               2.0       2.9            0
2  37.0              4.0               2.0       2.6            1
3   5.0              3.0               3.0       1.4            1
4  63.0              4.0               1.0       3.2            1

In [3]:
# Use the `describe` function to get the basic summary statistics
Thanksgiving.describe()


Ages  Dinner_Servings  Dessert_Servings   Duration  Macy_Parade
count  50.000000         50.00000         50.000000  50.000000    50.000000
mean   36.800000          3.78000          1.840000   2.400000     0.620000
std    15.841047          0.46467          0.680936   0.549954     0.490314
min     5.000000          3.00000          1.000000   1.300000     0.000000
25%    25.250000          4.00000          1.000000   2.025000     0.000000
50%    36.500000          4.00000          2.000000   2.500000     1.000000
75%    48.000000          4.00000          2.000000   2.800000     1.000000
max    74.000000          5.00000          3.000000   4.000000     1.000000

## Instructions:
Use this Thanksgiving dataset to explore patterns in the family dining habits.
You’ll need to _**interpret**_ the data using summary statistics, **not just report them**.

You will being using **only the describe function** to answer your questions

### Questions:

1. Looking at the minimum and maximum values of Age, what do they suggest about who attended this Thanksgiving dinner? <br>
**Answer:** Children to grandparents attended the dinner.<br>

2. Looking at the mean and standard deviation of Dinner_Duration, what do they show about how consistent the mealtime was across family members? <br>
**Answer:** Assuming hours with duration, with a mean of 2.4 they have a long dinner but with a standard deviation of 0.5499 the dinner could range with 1 standard deviation difference 1.8 to 3 hours. I assume the data shows the mealtime was fairly consistent.<br>

3. If someone stayed for 4.8 hours, would you consider them an outlier? _Use the mean and standard deviation to support your answer (Hint: $mean + 2*std$ OR $IQR=Q3−Q1$ and $Upper Bound=Q3+1.5×IQR$)._ <br>
**Answer:** Mean plus two standard deviation gets me 3.499, so 4.8 hours would be considered an outlier.<br>

4. Compare the mean and median of Dinner_Servings. What could a large difference between them indicate about family member behavior? _(Hint: 50% is half way in the data set)_ <br>
**Answer:** Multitude of family members favor less/more in their  servings.<br>

5. Looking at the max value of Dessert_Servings, what type of guest do you imagine that might be and what could this tell us about dessert consumption overall?  <br>
**Answer:** They might have somewhat of a sweet tooth. People like to have their desserts.<br>

6. Suppose the standard deviation of Dinner_Serving is small. What does that suggest about how much everyone ate?<br>
**Answer:** Everyone ate about the same amount of servings.<br>

7. Which statistic (min, max, mean, std) best helps you find the “light eaters”? Explain how you would identify them and if the minimum makes sense.<br>
**Answer:** Minimum, the minimum shows the least amount observed within the data. Using the minimum in dinner servings, I can determine around which are the light eaters.<br>

8. The range (max − min) of Dinner_Duration is quite large. What could that imply about how the dinner was organized?  <br>
**Answer:** People were allowed to eat however much they wanted, whether it be a lot or less.<br>

9. You suspect that older guests tend to eat fewer desserts. Could you use only the `describe` function to explore whether that idea might be true before doing a full analysis or would we have to include something else?  <br>
**Answer:** The describe function tells me there are older people but not any correlation between age and the idea they have fewer desserts. We would have to look at how much each person within an age group eats desserts within a graph as it can show a good representation of what we may be looking for.<br>

10. What does the mean tell you about how common it was for people to watch the Macy’s Day Parade?  <br>
**Answer:** The closer it is to one for the mean of this variable, the more people out of the total are watching the parade. As the mean is over 0.6, I would consider it was common for people to watch the parade.<br>

11. what does the standard deviation suggest about the variation in watching behavior? Was parade-watching behavior fairly consistent across people, or was there a lot of difference between individuals?  <br>
**Answer:** The standard deviation shows that all the data is within 1 to 2 standard deviations of the mean. Since there is only 2 variables I don't think there can be a lot of variance within this variable. On whether it was consistent, I think parade-watching behavior was consistent.<br>

12. In one sentence, summarize what kind of Thanksgiving this dataset represents. Focus on what the numbers imply rather than just stating the numbers.  <br>
**Answer:** A lot of the family gathered and enjoyed the meal while enjoying a thanksgiving parade and spending time together.<br>

13. Suggest some visuals that would be ideal for exploring this data and state their purpose. <br>
**Answer:** Several scatterplots to show how many persons are in each category which can help answer questions towards correlation and other questions.<br>
